### PR TITLE
nodePackages.terser: use buildNpmPackage

### PR DESCRIPTION
**Pull request by Lord-Valen**
## Description of changes
See https://github.com/NixOS/nixpkgs/issues/229475.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/terser/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ buildNpmPackage, fetchFromGitHub, lib, ... }:
```
Comment (natsukium) [inline pkgs/development/tools/misc/terser/default.nix:1]:
```suggestion
{ buildNpmPackage, fetchFromGitHub, lib}:
```

**Comment by natsukium** [comment]
@ofborg build terser

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 252499` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terser</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM

**Review by wegank — APPROVED** [review]

File: pkgs/development/tools/misc/terser/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    hash = "sha256-ZI5ElHnQwoCJspGL/v0PqddMUAAhQGWDZA9utWZD/nM=";
+  };
+
+  npmDepsHash = "sha256-M7LGXoZFBQrXpkiofnam7tgFkk6+N7ckPxTcwAAuqxU=";
+
+  meta = with lib; {
+    description = "JavaScript parser, mangler and compressor toolkit for ES6+";
+    homepage = "https://terser.org";
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ talyz ];
```
Comment (dotlambda) [inline pkgs/development/tools/misc/terser/default.nix:20]: Reference?

Comment (Lord-Valen) [reply pkgs/development/tools/misc/terser/default.nix:20]: https://github.com/NixOS/nixpkgs/pull/129668

**Review by dotlambda — APPROVED** [review]
Please fix the merge conflict.

**Comment by dotlambda** [comment]
@ofborg build terser

**Pull request merged**